plugins/obs-ffmpeg/obs-ffmpeg-source.c

```diff
@@ -386,6 +386,15 @@ static void ffmpeg_source_tick(void *data, float seconds)
 	}
 }
 
+#define SRT_PROTO "srt"
+#define RIST_PROTO "rist"
+
+static bool requires_mpegts(char *path)
+{
+	return !strncmp(path, SRT_PROTO, sizeof(SRT_PROTO) - 1) ||
```

Comment:
Is this always lowercase at this point, or will uppercase user input still be uppercase when it's checked here?

Reply:
The checks don't work for lowercase indeed. But the urls won't be recognized by ffmpeg in uppercase. We have the exact same check in ffmpeg-mux (for srt/rist outputs). I haven't seen any issues as yet with people using uppercase rather than lowercase so i'd say the check is safe.

Reply:
i don't think we do any check at all for strings to be uuper or lower case; it seems complicated to implement (ex: https://stackoverflow.com/questions/5820810/case-insensitive-string-comparison-in-c ) and germane to the problems i'm trying to fix. So i'd say until someone opens an issue because we don't handle correctly his url (be it rtmp, rist srt or else) because it's upper case, I'll leave this as it is

Reply:
For the record, we have case-insensitive comparison functions.

https://obsproject.com/docs/reference-libobs-util-dstr.html?highlight=astrcmpi#c.astrcmpi

However, if the expectation is that these are always lowercase and would fail in uppercase anyways, then that's another thing entirely, though I don't know if I agree with it.

Reply:
ok swapped the comp function to astrcmp_n.